pkg/skaffold/runner/build_deploy.go

```diff
@@ -92,6 +92,7 @@ func (r *SkaffoldRunner) DeployAndLog(ctx context.Context, out io.Writer, artifa
 	var imageNames []string
 	for _, artifact := range artifacts {
 		imageNames = append(imageNames, artifact.ImageName)
+		r.podSelector.Add(artifact.Tag)
```

Comment:
this is the fix, it's a one line change. Testing it is a bit more involved, hence the rest of the PR.